### PR TITLE
Double precision for BNG and Kappa params

### PR DESCRIPTION
**Pull request by alubbock**
As per #443, parameter values are not currently exported with full
precision to BNG and Kappa. This PR introduces 17 decimal digits,
in order to preserve full precision when round-tripping 64-bit
floats (as per IEEE 754 standard).

Since this might make previous simulations not exactly reproducible,
we might want to introduce a way of enabling users to use
reduced precision, either on a per-simulator or global level.
Open to suggestions on this. Might be complicated in the BNG case
by reaction network caching - local function parameters
read from BNG would need to be regenerated in the case
of a precision change.

Closes: #443

### CODE REVIEW

**Comment by jmuhlich** [comment]
I found that the same Docker container (thus identical system libraries), same PySB version, same model, same parameter sets, on **different machines** (one Linux, one Windows) gave slightly different results already, so I wouldn't be too worried about reproducibility here.

File: pysb/tests/test_importers.py [hunk 1]
```diff
@@ -16,10 +16,7 @@
 REDUCED_PRECISION = {
     'CaOscillate_Func': 1e-4,
     'michment': 1e-8,
-    'motor': 1e-8,
```
Comment (jmuhlich) [inline pysb/tests/test_importers.py]: Nice that this fixes most of the precision issues here!

**Comment by alubbock** [comment]
> I found that the same Docker container (thus identical system libraries), same PySB version, same model, same parameter sets, on **different machines** (one Linux, one Windows) gave slightly different results already, so I wouldn't be too worried about reproducibility here.

Strange. Same Docker engine version? Did you manage to isolate the cause?

I'll merge this in, but an explanation of the change should be included in the release notes.

**Comment by jmuhlich** [comment]
Didn't isolate the cause, but I know there are a few pieces of relevant FPU state that might not have been explicitly set by the code in the container or that can "leak" over from other processes outside the container during context switching. See more here:

https://randomascii.wordpress.com/2013/07/16/floating-point-determinism/

The differences were very small, but we were running a simulation to steady-state which is a case where tiny differences become readily noticeable. Even though we were using a tolerance for the convergence rather than waiting for all observables to be exactly equal for two steps, the two machines required an order of magnitude difference in simulated time (tspan) to settle to within that tolerance.

**Pull request merged**